timefold-solver-python-core/src/main/python/score/_group_by.py

```diff
@@ -993,6 +1013,120 @@ def to_sorted_map(key_mapper, value_mapper, merge_function_or_set_creator=None):
         else:
             raise ValueError
 
+    @overload
```

Comment:
These now give me `This decorator will not receive a callable it may expect; the built-in decorator returns a special object` warning. No other collector does that.

Reply:
All other collectors have `# noqa` after `overload for this reason. It tells the IDE everything is fine and no issues is caused by `@staticmethod` not returning a normal function. (TL;DR, staticmethods are objects that override `__get__` to do nothing, whereas normal functions `__get__` bind the instance to the first argument).